lib/api/actions_builder.rb

```diff
@@ -0,0 +1,93 @@
+module Api
+  class ActionsBuilder
+    attr_reader :config, :href, :type, :cspec, :collection, :subcollection, :resource
+
+    def initialize(request, href, type, resource = nil)
+      @config = CollectionConfig.new
```

Comment:
Similarly, you could pass this into the constructor (but maybe leaving it as the default), meaning you could test this with a fake config that conforms to the same structure. I think this would be OK since there should be good coverage that this works in general in the request specs, but would enable you to create edge cases that are not easy to find in the wild, and also decouples the tests from the specifics of the api.yml. Small changes there could cause these tests to fail, which would be unfortunate.